commands/operator-sdk/cmd/scorecard/basic_tests.go

```diff
@@ -0,0 +1,186 @@
+// status blocks exist or return an error after the timeout.
+func checkSpecAndStat(runtimeClient client.Client, obj unstructured.Unstructured, noStore bool) error {
+	testSpec := scorecardTest{testType: basicOperator, name: "Spec Block Exists", maximumPoints: 1}
+	testStat := scorecardTest{testType: basicOperator, name: "Status Block Exist", maximumPoints: 1}
+	var specPoints, statusPoints int
+	err := wait.Poll(time.Second*1, time.Second*time.Duration(SCConf.InitTimeout), func() (bool, error) {
+		err := runtimeClient.Get(context.TODO(), types.NamespacedName{Namespace: SCConf.Namespace, Name: name}, &obj)
+		if err != nil {
+			return false, fmt.Errorf("error getting custom resource: %v", err)
+		}
+		pass := true
+		if obj.Object["spec"] == nil {
+			pass = false
+			specPoints = 0
+		} else {
+			specPoints = 1
+		}
+		if obj.Object["status"] == nil {
+			pass = false
+			statusPoints = 0
+		} else {
+			statusPoints = 1
+		}
+		return pass, nil
+	})
+	if !noStore {
+		testSpec.earnedPoints = specPoints
+		testStat.earnedPoints = statusPoints
+		scTests = append(scTests, testSpec)
+		scTests = append(scTests, testStat)
+	}
+	if err != nil && !reflect.DeepEqual(err, wait.ErrWaitTimeout) {
+		return err
+	}
+	return nil
+}
+
+// TODO: user specified tests for operators
+
+// checkStatusUpdate looks at all fields in the spec section of a custom resource and attempts to modify them and
+// see if the status changes as a result. This is a bit prone to breakage as this is a black box test and we don't
+// know much about how the operators we are testing actually work and may pass an invalid value. In the future, we
+// should use user-specified tests
+func checkStatusUpdate(runtimeClient client.Client, obj unstructured.Unstructured) error {
+	test := scorecardTest{testType: basicOperator, name: "Operator actions are reflected in status", maximumPoints: 1}
+	err := runtimeClient.Get(context.TODO(), types.NamespacedName{Namespace: SCConf.Namespace, Name: name}, &obj)
+	if err != nil {
+		return fmt.Errorf("error getting custom resource: %v", err)
+	}
+	if obj.Object["status"] == nil || obj.Object["spec"] == nil {
+		scTests = append(scTests, test)
+		return nil
+	}
+	statCopy := make(map[string]interface{})
+	for k, v := range obj.Object["status"].(map[string]interface{}) {
+		statCopy[k] = v
+	}
+	pass := true
+	specMap := obj.Object["spec"].(map[string]interface{})
+	for k, v := range specMap {
+		switch t := v.(type) {
+		case int64:
+			specMap[k] = specMap[k].(int64) + 1
+		case float64:
+			specMap[k] = specMap[k].(float64) + 1
+		case string:
+			// TODO: try and find out how to make this better
+			// Since strings may be very operator specific, this test may not work correctly in many cases
+			specMap[k] = fmt.Sprintf("operator sdk test value %f", rand.Float64())
+		case bool:
+			specMap[k] = !specMap[k].(bool)
+		case []interface{}: // TODO: Decide how this should be handled
+		default:
```

Comment:
You can have nested objects, you should also handle `interface{}`

Reply:
I've added support for `map[string]interface{}` and `[]map[string]interface{}` and I now have both `interface{}` and `[]interface{}` as TODOs, as I'm not sure we can really do much with those.

Reply:
We shouldn't need an `interface{}` case right? `specMap`'s type is `map[string]interface{}`, and we perform a type switch on the `interface{}` value. That won't ever resolve back to `interface{}` will it?

Reply:
Yeah, I don't think that an `interface{}` can be type switched to only `interface{}`. I guess I could remove it. If this does somehow happen, that would just fall through to the default case of unknown type.

Reply:
You probably need to factor conversion into its own func and call it recursively once for each item in the slice of ‘[]interface{}’.